core/src/main/java/com/amplifyframework/auth/AuthCategoryBehavior.java

```diff
@@ -118,22 +118,68 @@ void confirmSignIn(
+    void signInWithSocialWebUI(
+            @NonNull AuthProvider provider,
             @NonNull Activity callingActivity,
-            @NonNull Consumer<String> onSuccess,
-            @NonNull Consumer<AmplifyException> onError);
+            @NonNull Consumer<AuthSignInResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Launch the specified auth provider's web UI sign in experience. You should also put the
+     * {@link #handleWebUISignInResponse(Intent)} method in your activity's onNewIntent method to
+     * capture the response which comes back from the UI flow.
+     * @param provider The auth provider you want to launch the web ui for (e.g. Facebook, Google, etc.)
+     * @param callingActivity The activity in your app you are calling this from
+     * @param options Advanced options for signing in with an auth provider's hosted web ui.
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void signInWithSocialWebUI(
+            @NonNull AuthProvider provider,
+            @NonNull Activity callingActivity,
+            @NonNull AuthWebUISignInOptions options,
+            @NonNull Consumer<AuthSignInResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Launch a hosted web sign in UI flow. You should also put the {@link #handleWebUISignInResponse(Intent)} method in
+     * your activity's onNewIntent method to capture the response which comes back from the UI flow.
+     * @param callingActivity The activity in your app you are calling this from
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void signInWithWebUI(
+            @NonNull Activity callingActivity,
+            @NonNull Consumer<AuthSignInResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Launch a hosted web sign in UI flow. You should also put the {@link #handleWebUISignInResponse(Intent)}
+     * method in your activity's onNewIntent method to capture the response which comes back from the UI flow.
+     * @param callingActivity The activity in your app you are calling this from
+     * @param options Advanced options for signing in with a hosted web ui.
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void signInWithWebUI(
+            @NonNull Activity callingActivity,
+            @NonNull AuthWebUISignInOptions options,
+            @NonNull Consumer<AuthSignInResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
 
     /**
-     * Handles the response which comes back from {@link #signInWithUI(Activity, Consumer, Consumer)}.
+     * Handles the response which comes back from {@link #signInWithWebUI(Activity, Consumer, Consumer)}.
      * @param intent The app activity's intent
      */
-    void handleSignInWithUIResponse(Intent intent);
+    void handleWebUISignInResponse(Intent intent);
```

Comment:
why are we deviating from `SignInWith<>` pattern here? Can't this method just be `handleSignInWithWebUIResponse`?

Reply:
This was the decision made in the naming meeting so just rolling with it